From 2ede92fbe4ad523334bea6f2f0418f004860b68c Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Wed, 04 Jun 2025 19:39:33 +0300
Subject: [PATCH] [JSW-824] jgs: added files for onix build

Added board configurations and device tree files for
JGS onix build.

Change-Id: I73e43724036f6141f635dc82a4112dd58afa620d
Signed-off-by: Shivani Raval <sraval@habana.ai>
---

diff --git a/boards/intel/jgs/CMakeLists.txt b/boards/intel/jgs/CMakeLists.txt
index bc5027a..5ad5f81 100644
--- a/boards/intel/jgs/CMakeLists.txt
+++ b/boards/intel/jgs/CMakeLists.txt
@@ -3,8 +3,12 @@
 
 # TODO: use pinctrl driver for pinmux
 zephyr_sources_ifdef(CONFIG_BOARD_JGS_PREBOOT_ASIC pinmux_common.c)
+zephyr_sources_ifdef(CONFIG_BOARD_JGS_PREBOOT_ONIX pinmux_common.c)
 
 zephyr_sources_ifdef(CONFIG_BOARD_JGS_RST_ASIC pinmux_common.c)
+zephyr_sources_ifdef(CONFIG_BOARD_JGS_RST_ONIX pinmux_common.c)
 
 zephyr_sources_ifdef(CONFIG_BOARD_JGS_MGMT_ASIC pinmux_common.c)
+zephyr_sources_ifdef(CONFIG_BOARD_JGS_MGMT_ONIX pinmux_common.c)
+
 
diff --git a/boards/intel/jgs/Kconfig.jgs_mgmt_onix b/boards/intel/jgs/Kconfig.jgs_mgmt_onix
new file mode 100644
index 0000000..68e530f
--- /dev/null
+++ b/boards/intel/jgs/Kconfig.jgs_mgmt_onix
@@ -0,0 +1,5 @@
+# Copyright 2022 HabanaLabs, Ltd. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_JGS_MGMT_ONIX
+	select SOC_JGS_MGMT
diff --git a/boards/intel/jgs/Kconfig.jgs_preboot_onix b/boards/intel/jgs/Kconfig.jgs_preboot_onix
new file mode 100644
index 0000000..3c3e610
--- /dev/null
+++ b/boards/intel/jgs/Kconfig.jgs_preboot_onix
@@ -0,0 +1,5 @@
+# Copyright 2022 HabanaLabs, Ltd. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_JGS_PREBOOT_ONIX
+	select SOC_JGS_PREBOOT
diff --git a/boards/intel/jgs/Kconfig.jgs_rst_onix b/boards/intel/jgs/Kconfig.jgs_rst_onix
new file mode 100644
index 0000000..d0cd96f
--- /dev/null
+++ b/boards/intel/jgs/Kconfig.jgs_rst_onix
@@ -0,0 +1,5 @@
+# Copyright 2022 HabanaLabs, Ltd. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_JGS_RST_ONIX
+	select SOC_JGS_RST
diff --git a/boards/intel/jgs/board.yml b/boards/intel/jgs/board.yml
index 83fe66c..4c7d9dc 100644
--- a/boards/intel/jgs/board.yml
+++ b/boards/intel/jgs/board.yml
@@ -11,3 +11,21 @@
     vendor: intel
     socs:
       - name: mgmt
+  - name: jgs_preboot_onix
+    vendor: intel
+    socs:
+      - name: preboot
+  - name: jgs_mgmt_onix
+    vendor: intel
+    socs:
+      - name: mgmt
+  - name: jgs_rst_onix
+    vendor: intel
+    socs:
+      - name: rst
+
+
+
+
+
+
diff --git a/boards/intel/jgs/jgs_mgmt_onix.dts b/boards/intel/jgs/jgs_mgmt_onix.dts
new file mode 100644
index 0000000..21fbff9
--- /dev/null
+++ b/boards/intel/jgs/jgs_mgmt_onix.dts
@@ -0,0 +1,19 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+/dts-v1/;
+
+#include <jgs_io_mgmt.dtsi>
+
+&refclk {
+	clock-frequency = <35000000>;
+};
+
+&uart0 {
+	current-speed = <57600>;
+	status = "okay";
+};
diff --git a/boards/intel/jgs/jgs_mgmt_onix_defconfig b/boards/intel/jgs/jgs_mgmt_onix_defconfig
new file mode 100644
index 0000000..bb49816
--- /dev/null
+++ b/boards/intel/jgs/jgs_mgmt_onix_defconfig
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_CPU_ARCHS_NO_HW_ATOMIC=y
+#CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE=2048
+CONFIG_BUILD_OUTPUT_BIN=y
+CONFIG_MAIN_STACK_SIZE=5120
+CONFIG_PRINTK=y
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=y
+CONFIG_LOG_MAX_LEVEL=4
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_EMPTY_WEAK_FUNCTIONS=y
+CONFIG_LINKER_ORPHAN_SECTION_PLACE=y
+CONFIG_XIP=n
+CONFIG_HEAP_MEM_POOL_SIZE=4096
+CONFIG_HL_EFUSE=y
diff --git a/boards/intel/jgs/jgs_preboot_onix.dts b/boards/intel/jgs/jgs_preboot_onix.dts
new file mode 100644
index 0000000..4d44f27
--- /dev/null
+++ b/boards/intel/jgs/jgs_preboot_onix.dts
@@ -0,0 +1,23 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2023 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+/dts-v1/;
+
+#include <jgs_preboot.dtsi>
+
+/ {
+	model = "jgs_preboot_d1";
+};
+
+&refclk {
+	clock-frequency = <35000000>;
+};
+
+&uart0 {
+	current-speed = <57600>;
+	status = "okay";
+};
diff --git a/boards/intel/jgs/jgs_preboot_onix_defconfig b/boards/intel/jgs/jgs_preboot_onix_defconfig
new file mode 100644
index 0000000..bfd9c67
--- /dev/null
+++ b/boards/intel/jgs/jgs_preboot_onix_defconfig
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_CPU_ARCHS_NO_HW_ATOMIC=y
+CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE=2048
+CONFIG_BUILD_OUTPUT_BIN=y
+CONFIG_MAIN_STACK_SIZE=5120
+CONFIG_PRINTK=y
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=y
+CONFIG_LOG_MAX_LEVEL=4
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+CONFIG_MBEDTLS=y
+CONFIG_MBEDTLS_BUILTIN=y
+CONFIG_MBEDTLS_SHA384=y
+CONFIG_LIBFDT=y
+CONFIG_LIBBOOTINFO=y
+# boot info addr is spsram base
+CONFIG_BOOTINFO_ADDR=0x78a00000
+CONFIG_BOOTINFO_SIZE=0x8000
+CONFIG_LIBSOC_INIT=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_LINKER_ORPHAN_SECTION_PLACE=y
+CONFIG_FAULT_THREAD_DUMP=y
+CONFIG_HEAP_MEM_POOL_SIZE=3072
+CONFIG_MINIMAL_LIBC_OPTIMIZE_STRING_FOR_SIZE=n
+CONFIG_BOOT_IF_PSEUDO_RO_TIMER=y
+CONFIG_HL_EFUSE=y
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=35000000
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=10000
+CONFIG_EMPTY_WEAK_FUNCTIONS=y
+CONFIG_MAIN_THREAD_PRIORITY=4
+CONFIG_ARCV2_INTERRUPT_UNIT=y
+CONFIG_ARCV2_TIMER=y
+CONFIG_POLL=y
+CONFIG_THREAD_NAME=y
+CONFIG_XIP=n
+CONFIG_CRC=y
diff --git a/boards/intel/jgs/jgs_rst_onix.dts b/boards/intel/jgs/jgs_rst_onix.dts
new file mode 100644
index 0000000..dba498e
--- /dev/null
+++ b/boards/intel/jgs/jgs_rst_onix.dts
@@ -0,0 +1,29 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+/dts-v1/;
+
+#include <jgs_rst.dtsi>
+
+/ {
+	model = "jgs_rst";
+	compatible = "intel,jgs_rst";
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+	};
+};
+
+&refclk {
+	clock-frequency = <35000000>;
+};
+
+&uart0 {
+	current-speed = <57600>;
+	status = "okay";
+};
diff --git a/boards/intel/jgs/jgs_rst_onix_defconfig b/boards/intel/jgs/jgs_rst_onix_defconfig
new file mode 100644
index 0000000..acec5ae
--- /dev/null
+++ b/boards/intel/jgs/jgs_rst_onix_defconfig
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_CPU_ARCHS_NO_HW_ATOMIC=y
+CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE=2048
+CONFIG_BUILD_OUTPUT_BIN=y
+CONFIG_MAIN_STACK_SIZE=5120
+CONFIG_PRINTK=y
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=y
+CONFIG_LOG_MAX_LEVEL=4
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_EMPTY_WEAK_FUNCTIONS=y
+CONFIG_LINKER_ORPHAN_SECTION_PLACE=y
+CONFIG_MINIMAL_LIBC_OPTIMIZE_STRING_FOR_SIZE=n
+CONFIG_BOOT_IF_PSEUDO_RO_TIMER=y
+CONFIG_XIP=n
